src/ipaddr.coffee

```diff
@@ -381,7 +381,7 @@ expandIPv6 = (string, parts) ->
     return null
 
   # Remove zone index and save it for later
-  zoneId = (string.match(zoneIndex) || [])[0]
+  zoneId = (string.match(ipv6Regexes['zoneIndex']) || [])[0]
```

Comment:
No worries, it took me a while to figure out how it was working in the first place for the other cases. 😝